ee/clickhouse/queries/person_query.py

```diff
@@ -1,36 +1,49 @@
-from typing import Optional, Union
+from typing import List, Optional, Union
 
+from ee.clickhouse.materialized_columns.columns import ColumnName
 from ee.clickhouse.queries.column_optimizer import ColumnOptimizer
 from posthog.models import Filter
 from posthog.models.filters.path_filter import PathFilter
 
 
 class ClickhousePersonQuery:
     PERSON_PROPERTIES_ALIAS = "person_props"
+    ALIASES = {"properties": "person_props"}
 
     _filter: Union[Filter, PathFilter]
     _team_id: int
     _column_optimizer: ColumnOptimizer
+    _extra_fields: List[ColumnName]
 
     def __init__(
-        self, filter: Union[Filter, PathFilter], team_id: int, column_optimizer: Optional[ColumnOptimizer] = None
+        self,
+        filter: Union[Filter, PathFilter],
+        team_id: int,
+        column_optimizer: Optional[ColumnOptimizer] = None,
+        extra_fields: List[ColumnName] = [],
     ) -> None:
         self._filter = filter
         self._team_id = team_id
         self._column_optimizer = column_optimizer or ColumnOptimizer(self._filter, self._team_id)
+        self._extra_fields = extra_fields
 
     def get_query(self) -> str:
         fields = (
             "id"
             + (
                 f", argMax(properties, _timestamp) AS {self.PERSON_PROPERTIES_ALIAS}"
                 if self._column_optimizer.should_query_person_properties_column
+                or self.PERSON_PROPERTIES_ALIAS in self._extra_fields
                 else ""
             )
             + " ".join(
                 f", argMax({column_name}, _timestamp) as {column_name}"
                 for column_name in self._column_optimizer.materialized_person_columns_to_query
             )
+            + " ".join(
+                f", argMax({column_name}, _timestamp) as {self.ALIASES.get(column_name, column_name)}"
```

Comment:
I gather the argMax is simulating something like a DISTINCT ON(id) query? Is there any reason we don't use this instead?

Reply:
https://clickhouse.tech/docs/en/sql-reference/aggregate-functions/reference/argmax/

Since there can be multiple rows with the same id (due to updates), we need to select latest properties. DISTINCT would not help us beyond getting the id.

> Calculates the arg value for a maximum val value. If there are several different values of arg for maximum values of val, returns the first of these values encountered.

This is what we need here to get the "latest" column value for this person from CH.